### PR TITLE
add ability to tweak number of lines read for read_table

### PR DESCRIPTION
**Pull request by Yeedle**
Documents and "fixes" [issue #518](https://github.com/tidyverse/readr/issues/518).

### CODE REVIEW

**Comment by Yeedle** [comment]
On second thought, perhaps this shouldn't be exposed to the user. But it certainly should be documented.

File: R/read_fwf.R [hunk 1]
```diff
@@ -62,12 +62,14 @@ read_fwf <- function(file, col_positions, col_types = NULL,
   warn_problems(out, source_name(file))
 }
 
-#' @rdname read_fwf
+#' @rdname read_fw
```
Comment (hadley) [inline R/read_fwf.R]: Typo?

File: R/read_fwf.R [hunk 2]
```diff
@@ -62,12 +62,14 @@ read_fwf <- function(file, col_positions, col_types = NULL,
...
 #' @export
-fwf_empty <- function(file, skip = 0, col_names = NULL, comment = "") {
+#' @param n Number of lines the tokenizer will read to determine file structure. By default
+#'      it is set to 100.
+fwf_empty <- function(file, skip = 0, col_names = NULL, comment = "", n = 100L) {
   ds <- datasource(file, skip = skip)
 
-  out <- whitespaceColumns(ds, comment = comment)
+  out <- whitespaceColumns(ds, comment = comment, n=n)
```
Comment (hadley) [inline R/read_fwf.R]: Please put spaces around `=`

File: R/read_table.R [hunk 3]
```diff
@@ -29,8 +31,9 @@
 read_table <- function(file, col_names = TRUE, col_types = NULL,
                        locale = default_locale(), na = "NA", skip = 0,
                        n_max = Inf, guess_max = min(n_max, 1000),
-                       progress = interactive()) {
-  columns <- fwf_empty(file, skip = skip)
+                       progress = interactive(),
```
Comment (hadley) [inline R/read_table.R]: The indenting looks messed up here.

Comment (krlmlr) [inline R/read_table.R]: Indenting looks fine to me now.

Comment (Yeedle) [inline R/read_table.R]: I fixed it, as well as the other issues @hadley pointed out.

File: R/read_table.R [hunk 4]
```diff
@@ -14,6 +14,8 @@
 #' @inheritParams datasource
 #' @inheritParams tokenizer_fwf
 #' @inheritParams read_delim
+#' @param n The number of lines the tokenizer will read to determine the structure
```
Comment (hadley) [inline R/read_table.R]: Should this be `n_max`?

Comment (hadley) [inline R/read_table.R]: Oops no,

**Comment by hadley** [comment]
Looks good. If you could add a bullet to NEWS, we'll be ready to merge :smile:

**Comment by Yeedle** [comment]
Added the bullet. Formatting in NEWS was messed up by previous commit. Hope that versioning of news is still correct.

**Pull request merged**